BusinessLogic/Services/EmployeeService.cs

```diff
@@ -1,16 +1,104 @@
-﻿using BusinessLogic.IRepositories;
+﻿using BusinessLogic.BusinessRules;
+using BusinessLogic.Interfaces;
+using BusinessLogic.IRepositories;
 using BusinessLogic.Models;
+using BusinessLogic.Projections;
+using UserInterface.Stores;
 
 namespace BusinessLogic.Services;
 
 public class EmployeeService
 {
     private IEmployeeRepository employeeRepository;
+    private IEmployeeLoginRepository employeeLoginRepository;
+    private IHashingService hashingService;
 
-    public EmployeeService(IEmployeeRepository employeeRepository) => this.employeeRepository = employeeRepository;
+    public EmployeeService(IEmployeeRepository employeeRepository, IEmployeeLoginRepository employeeLoginRepository, IHashingService hashingService) : this(employeeRepository, employeeLoginRepository)
+    {
+        this.hashingService = hashingService;
+    }
+
+    public EmployeeService(IEmployeeRepository employeeRepository, IEmployeeLoginRepository employeeLoginRepository) : this(employeeRepository)
+    {
+        this.employeeLoginRepository = employeeLoginRepository;
+    }
 
+    public EmployeeService(IEmployeeRepository employeeRepository)
+    {
+        this.employeeRepository = employeeRepository;
+    }
     public Employee GetEmployeeById(Guid id)
     {
         return employeeRepository.GetEmployeeById(id);
     }
+    public List<EmployeeProjection> GetEmployeesProjections()
+    {
+        return employeeRepository.GetAllEmployeeProjections();
+    }
+
+    private void SaveEmployee(Employee employee)
+    {
+        employeeRepository.SaveEmployee(employee);
```

Comment:
waar is de validatie in deze methods? wat als de frontend wijzigt?

Reply:
Zie "ProcessEmployee" functie